flink-tests/src/test/java/org/apache/flink/test/checkpointing/CheckpointAfterAllTasksFinishedITCase.java

```diff
@@ -82,7 +83,8 @@ public void setUp() {
     @Test
     public void testImmediateCheckpointing() throws Exception {
         env.setRestartStrategy(RestartStrategies.noRestart());
-        env.enableCheckpointing(Long.MAX_VALUE - 1);
+        env.enableCheckpointing(
+                Duration.ofNanos(Long.MAX_VALUE /* max allowed by FLINK */).toMillis());
```

Comment:
Because this pr make the env.enableCheckpointing updates the env's configuration. This configuration is utilized as the cluster config when using LocalExecutor. Consequently, there are places where the toMap method of the configuration is invoked, such as within the constructor of ClusterConfigHandler.

However, in Flink the conversion between Duration and String requires that the maximum value for Duration corresponds to the number of nanoseconds in Long.MAX_VALUE. Thus, this change was made to adhere to this requirement.

Reply:
I see. What confused me is that it changes `Long.MAX_VALUE - 1` to `Long.MAX_VALUE`. I think these two numbers do not make actual differences though.
Besides that, I can see other usages of `enableCheckpointing(...)` which passes in a large number, e.g. `CheckpointAfterAllTasksFinishedITCase#testRestoreAfterSomeTasksFinished()` and `RestoreUpgradedJobITCase#runOriginalJob()`.

Reply:
1. The previous use of Long.MAX_VALUE - 1 as the checkpoint interval was intentional because using Long.MAX_VALUE itself would effectively disable checkpointing, which isn't the desired behavior. The intention was to enable checkpointing with an interval so large that it effectively prevents any checkpoint from being triggered during the test run. The use of Duration.ofNanos(Long.MAX_VALUE).toMillis() aligns well with this scenario.
2. I will review all instances where a large checkpoint interval is used and add comments to clarify the reason using these intervals.